### PR TITLE
Use quick-xml instead of xml-rs

### PR DESCRIPTION
**Pull request by Erk-**
Since XML-rs is for the time being unmaintained[^1], and have been
marked as such in RUSTSEC[^2]. I have ported the XML parsing to
quick-xml.

Supersedes #107 

[^1]: https://github.com/netvl/xml-rs/issues/221

[^2]: https://rustsec.org/advisories/RUSTSEC-2022-0048

### CODE REVIEW

File: build/parser.rs [hunk 1]
```diff
@@ -1036,24 +1037,24 @@ pub enum MavXmlElement {
...
-        "entry" => Some(Entry),
-        "description" => Some(Description),
-        "param" => Some(Param),
-        "messages" => Some(Messages),
-        "message" => Some(Message),
-        "field" => Some(Field),
-        "deprecated" => Some(Deprecated),
-        "wip" => Some(Wip),
-        "extensions" => Some(Extensions),
+        b"version" => Some(Version),
```
Comment (patrickelectric) [inline build/parser.rs:1043]: Isn't it better to transform the xml access to utf-8 over than using all our strings to bytes ?

Comment (Erk-) [reply build/parser.rs:1043]: Do you mean to convert it to strings when we compare, or do you mean to use the Decoder of quick-xml/unescape_and_decode?

quick-xml does all access by default in bytes, i tried to only convert it to strings where it is needed.

Comment (patrickelectric) [reply build/parser.rs:1043]: Ok, I think that we continue with the current version of the PR.

**Review by patrickelectric — CHANGES_REQUESTED** [review]
CI is failing, I would recommend to check and test the CI in your own fork. Right now I need to approve the CI in the repository everytime that you update the PR.

**Comment by Erk-** [comment]
> CI is failing, I would recommend to check and test the CI in your own fork. Right now I need to approve the CI in the repository everytime that you update the PR.

Seems it was only the fmt, it looks to get through CI now at least https://github.com/fusion-engineering-forks/rust-mavlink/runs/7946114295

**Pull request merged**